### PR TITLE
Apply gh-5553 to io.micrometer.core.instrument.binder.jdk

### PR DESCRIPTION
**Pull request by izeye**
This PR applies the changes made in gh-5553 to the classes in the `io.micrometer.core.instrument.binder.jdk` package.

### CODE REVIEW

**Review by shakuzen — APPROVED** [review]

**Review by jonatan-ivanov — CHANGES_REQUESTED** [review]

File: micrometer-core/src/test/java11/io/micrometer/core/instrument/binder/jdk/MicrometerHttpClientTests.java [hunk 1]
```diff
@@ -84,6 +91,21 @@ void shouldInstrumentHttpClientWithTimer(WireMockRuntimeInfo wmInfo) throws IOEx
...
 
+    @Test
+    void shouldThrowErrorFromSendAsync(WireMockRuntimeInfo wmInfo) {
+        var client = MicrometerHttpClient.instrumentationBuilder(httpClient, meterRegistry).build();
+        var request = HttpRequest.newBuilder(URI.create(wmInfo.getHttpBaseUrl() + "/test-fault")).GET().build();
+        var response = client.sendAsync(request, HttpResponse.BodyHandlers.ofString());
+        assertThatThrownBy(response::join).isInstanceOf(CompletionException.class);
+        assertThatNoException().isThrownBy(() -> meterRegistry.get("http.client.requests")
+            .tag("method", "GET")
+            // TODO fix status/uri in exceptional cases where response may be null
```
[the remark this answers is not in the document]
Comment (izeye) [reply micrometer-core/src/test/java11/io/micrometer/core/instrument/binder/jdk/MicrometerHttpClientTests.java:102]: @jonatan-ivanov I just copied it from 81b53d3163e951e9c7abd4a4b8a9109927d6e9d7, and actually I made some polishing on it in https://github.com/micrometer-metrics/micrometer/pull/5705, but I didn't apply it here as I was sure about it.

File: micrometer-core/src/main/java11/io/micrometer/core/instrument/binder/jdk/MicrometerHttpClient.java [hunk 2]
```diff
@@ -272,11 +265,12 @@ public <T> CompletableFuture<HttpResponse<T>> sendAsync(HttpRequest httpRequest,
                 httpRequestBuilder);
         HttpRequest request = httpRequestBuilder.build();
         return client.sendAsync(request, bodyHandler, pushPromiseHandler).handle((response, throwable) -> {
+            instrumentation.setResponse(response);
+            stopObservationOrTimer(instrumentation, request, response);
```
[the remark this answers is not in the document]
Comment (izeye) [reply micrometer-core/src/main/java11/io/micrometer/core/instrument/binder/jdk/MicrometerHttpClient.java:269]: @jonatan-ivanov Sorry, I'll fix it and add a test shortly.

Actually I made the same mistake in https://github.com/micrometer-metrics/micrometer/pull/5708. I'll create another PR to fix it.

[the remark this answers is not in the document]
Comment (izeye) [reply micrometer-core/src/main/java11/io/micrometer/core/instrument/binder/jdk/MicrometerHttpClient.java]: > Actually I made the same mistake in #5708. I'll create another PR to fix it.

@jonatan-ivanov I created https://github.com/micrometer-metrics/micrometer/pull/5709 to fix it.

**Comment by izeye** [comment]
@jonatan-ivanov Thanks for the feedback! I applied your suggestions.

**Review by jonatan-ivanov — APPROVED** [review]

**Comment by jonatan-ivanov** [comment]
Thank you!

**Pull request merged**